src/components/ui/TaskTable.tsx

```diff
@@ -0,0 +1,209 @@
+    }
+  };
+
+  const handleReorder = async (reorderedTasks: Task[]) => {
+    try {
+      const updatedTasks = reorderedTasks.map((task, index) => ({
+        ...task,
+        position: index,
+      }));
+
+      setTaskList(updatedTasks);
+
+      for (const task of updatedTasks) {
+        await updateTaskMutation({
+          variables: {
+            input: {
+              id: task.id,
+              position: task.position,
+            },
+          },
+        });
+      }
+    } catch (error) {
+      console.error("Failed to reorder tasks:", error);
+    }
+  };
+
+  const [taskListRef, _taskItems, setTaskItems] = useDragAndDrop<HTMLTableSectionElement, Task>(taskList, {
+    group: "taskList",
+    dragHandle: ".kanban-handle",
+    onDragend: async (selectedTask) => {
+      const draggedTask = selectedTask.draggedNode.data.value as Task;
+      const newStatus = selectedTask.parent?.el?.id as Status;
+
+      if (newStatus && draggedTask.status !== newStatus) {
+        await handleDrop(draggedTask, newStatus);
+      } else {
+        await handleReorder(selectedTask.values as Task[]);
+      }
+    },
+  });
+
+  useEffect(() => {
+    setTaskList(tasks ?? []);
+    setTaskItems(tasks ?? []);
+  }, [tasks, status]);
+
+  return (
+    <Accordion className="w-full min-w-[348px]" collapsible type="single">
+      <AccordionItem className="accordion-item" value="item-1">
+        <AccordionTrigger className="AccordionTrigger flex h-14 w-full items-center justify-start gap-2 rounded-t border border-color_neutral_3 bg-color_neutral_4 pl-2 text-body-L font-bold text-color_neutral_1">
+          <ChevronDownIcon
+            className={
+              "AccordionChevron text-color_neutral_2 transition-transform"
+            }
+            aria-hidden
+          />
+          {formatStatus(status)}{" "}
+          <p className="text-color_neutral_2">({tasks?.length})</p>
+        </AccordionTrigger>
+        <AccordionContent className="accordion-content">
+          <table className="h-auto w-full border-collapse border-spacing-x-0 border-spacing-y-[15px] rounded-xl text-justify">
+            <tbody ref={taskListRef} className="flex flex-col bg-color_neutral_4 text-body-M text-color_neutral_1">
+              {taskList?.length !== 0 ? (
+                taskList?.map((task, index) => (
+                  <tr
+                    className="flex h-14 w-full items-center justify-between text-body-M text-color_neutral_1 last:rounded-ee-lg last:rounded-es-lg"
+                    key={task.id}
+                  >
+                    <td className="relative flex h-14 w-2/5 min-w-80 items-center gap-3 border border-color_neutral_3 pl-4">
+                      <span className="drag-handle">
+                        <GridHorizontalIcon className="text-color_neutral_2" />
+                      </span>
+                      <span
+                        className={cn("absolute left-0 h-[80%] w-1", {
```

Comment:
what about on extract this using properly the helper function

const dueDate = new Date(task.dueDate);
const colorClass = getDueDateColor(dueDate);

const backgroundClass = cn(
  "absolute left-0 h-[80%] w-1",
  {
   // this in an example you can improve it to match your requirements
    [`bg-${colorClass}`]: ["red", "yellow", "green"].includes(colorClass),
  }
);